photon-api/src/main/scala/com/linkedin/photon/ml/function/ObjectiveFunctionHelper.scala

```diff
@@ -26,7 +26,9 @@ import com.linkedin.photon.ml.optimization.game.CoordinateOptimizationConfigurat
  */
 object ObjectiveFunctionHelper {
 
-  type ObjectiveFunctionFactory = (CoordinateOptimizationConfiguration) => ObjectiveFunction
+  type ObjectiveFunctionFactoryFactory = CoordinateOptimizationConfiguration => Option[Int] => ObjectiveFunction
```

Comment:
Do we need to add parentheses for readability?

Reply:
It's a style preference question, I suppose. We can go through later and make sure that we're being consistent. IDEA complains about unnecessary parentheses if we include them, which is why I removed them.